forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/repository/migration/v3_1_6/MongoDbPaymentsChangeLog.java

```diff
@@ -24,83 +24,215 @@
+        Update update = new Update();
+        List<Pair<Query, Update>> updates = new ArrayList<>();
+        mongoTemplate.find(query, legacyClazz).forEach(
+                legacyClassDocInstance -> {
+                    try {
+                        // return a Optional Pair [id, countrySubDivision] for each document
+                        Optional<Pair<String, String>> optIdAndCountrySubDivision = getPair(legacyClazz, legacyClassDocInstance, writeRiskParentField);
+                        if (optIdAndCountrySubDivision.isPresent()) {
+                            if (optIdAndCountrySubDivision.get().getFirst() != null && optIdAndCountrySubDivision.get().getSecond() != null) {
+                                // we get only the first value of countrySubDivisions
+                                Query queryUpdate = new Query().addCriteria(new Criteria("_id").is(optIdAndCountrySubDivision.get().getFirst()));
+                                updates.add(Pair.of(queryUpdate, update.set(jsonPathCriteria, optIdAndCountrySubDivision.get().getSecond())));
+                            }
+                        }
+                    } catch (IntrospectionException | InvocationTargetException | IllegalAccessException e) {
+                        log.error("Error getting the id and countrySubDivision from the document [{}]", legacyClassDocInstance.toString(), e);
+                    }
+                }
+        );
+        // run the bulk operation
+        if (!updates.isEmpty()) {
+            docsUpdated = bulkOperations.updateMulti(updates).execute().getModifiedCount();
+        }
+        // execution time control - stop
+        stop(legacyClazz, docsUpdated, elapsedTime);
+        return docsUpdated;
     }
 
-    private void migrateInternationalPayments(MongoTemplate mongoTemplate) {
-        CloseableIterator<FRInternationalConsent2> frInternationalConsents = getLegacyDocuments(mongoTemplate, FRInternationalConsent2.class);
-        frInternationalConsents.forEachRemaining(d -> migrate(mongoTemplate, d, toFRInternationalConsent(d)));
+    /*
+     * Introspect the legacyClassDocInstance to get the id and countrySubDivision values to prepare the bulk update operation
+     * @param legacyClazz collection class
+     * @param legacyClassDocInstance document legacyClass instance
+     * @param writeRiskParentField field name to get the write object that contains the risk object related with the collection class
+     * @param <T> generic parameter to conforms legacyClazz object
+     * @return a optional pair with the id and the first value of countrySubDivision
+     * @throws IntrospectionException
+     * @throws InvocationTargetException
+     * @throws IllegalAccessException
+     */
+    private <T> Optional<Pair<String, String>> getPair(Class<T> legacyClazz, Object legacyClassDocInstance, String writeRiskParentField) throws IntrospectionException, InvocationTargetException, IllegalAccessException {
```

Comment:
This isn't a very descriptive method name. I know it gets a `Pair`, but that's just the underlying classname (a bit like having a method called `getMap()` or `getString()`).

Perhaps `getDocumentIdAndCountrySubDivision()` would be better?

Reply:
Method deleted.